ner_v2/detectors/temporal/date/en/date_detection.py

```diff
@@ -1660,28 +1660,35 @@ def _date_range_ddth_of_mmm_to_ddth(self, date_list=None, original_list=None):
         if date_list is None:
             date_list = []
         regex_pattern = re.compile(r'\b(([12][0-9]|3[01]|0?[1-9])\s?(?:nd|st|rd|th)?\s+(?:of\s+)?([A-Za-z]+)\s+'
-                                   r'(?:-|to|-|till)\s+([12][0-9]|3[01]|0?[1-9])\s?(?:nd|st|rd|th)?)\b')
+                                   r'(?:-|to|-|till)\s+([12][0-9]|3[01]|0?[1-9])\s?(?:nd|st|rd|th)?\s?([a-zA-Z]+)?)\b')
         patterns = regex_pattern.findall(self.processed_text.lower())
         for pattern in patterns:
             original = pattern[0]
             dd1 = pattern[1]
             dd2 = pattern[3]
-            probable_mm = pattern[2]
-            mm = self.__get_month_index(probable_mm)
-            yy = self.now_date.year
-            if mm:
-                if self.now_date.month > int(mm):
-                    yy += 1
+            probable_mm1 = pattern[2]
+            probable_mm2 = pattern[4]
+            mm1 = self.__get_month_index(probable_mm1)
+            mm2 = mm1
+            if probable_mm2:
+                mm2 = self.__get_month_index(probable_mm2)
+            yy1 = self.now_date.year
+            yy2 = self.now_date.year
+            if mm1 and mm2:
+                if self.now_date.month > int(mm1):
+                    yy1 += 1
+                if self.now_date.month > int(mm2):
+                    yy2 += 1
```

Comment:
What happens when we get something like `2nd Jan to 3rd Feb 2020` 
we are disregarding the year here. Just check once what happens currently on develop/master